hack/e2e-common.sh

```diff
@@ -133,8 +133,14 @@ if $USE_MANAGED_DNS; then
 	# This is to prevent name conflicts across customer clusters.
 	CLUSTER_SHARD=$(cat /dev/urandom | tr -dc 'a-z' | fold -w 8 | head -n 1)
 	CLUSTER_DOMAIN="${CLUSTER_SHARD}.${BASE_DOMAIN}"
-	go run "${SRC_ROOT}/contrib/cmd/hiveutil/main.go" adm manage-dns enable ${BASE_DOMAIN} \
-		--creds-file="${CREDS_FILE}" --cloud="${CLOUD}"
+	# We've seen this failing 409 (concurrent updates) recently. So try a couple times.
+	# TODO: Check output for "the object has been modified" and only retry if it matches.
+	echo "enabling managed DNS"
+	for i in 1 2 3; do
+	  go run "${SRC_ROOT}/contrib/cmd/hiveutil/main.go" adm manage-dns enable ${BASE_DOMAIN} \
+	    --creds-file="${CREDS_FILE}" --cloud="${CLOUD}" && break
+	  echo "Retrying..."
```

Comment:
should we put a small sleep here?

Reply:
Shouldn't be necessary. The reason the 409 happens is because some update comes in between the GET and the POST within the execution of this single command.

I guess a sleep would help if we think _multiple_ updates are happening and need a few seconds to settle. WDYT?

Reply:
It's probably safe to just do the 3 retries. If we're competing with anything it would be an in-cluster controller, and hopefully we don't have controllers that are doing multiple writes during a single reconciliation. I'm fine either way. Let me know and i'll lgtm it

Reply:
Yeah, let's just go with it. I saw this error a couple of times on a couple of PRs over a couple of days, but haven't seen it since, so it's clearly not a high frequency thing in any case.